lib/src/model/package_builder.dart

```diff
@@ -192,7 +192,9 @@ class PubPackageBuilder implements PackageBuilder {
         for (var export in element.libraryExports) {
           _addKnownFiles(export.exportedLibrary);
         }
-        for (var part in element.parts2) {
+        for (var part in element.parts2
+            .map((e) => e.uri)
+            .whereType<DirectiveUriWithUnit>()) {
           _knownFiles.add(part.source.fullName);
```

Comment:
No, here `part.source` would be the source where the `part` (type `PartElement`) directive is located, i.e. the library source, not the source of the included unit.

You need `element.parts2.map((e) => e.uri).whereType<DirectiveUriWithUnit>().map((uri) => uri.unit)` to get included `CompilationUnitElement`s.

Reply:
Thanks!